app/scenes/image_preview/image_preview.js

```diff
@@ -0,0 +1,320 @@
+                        horizontal={true}
+                        pagingEnabled={this.state.pagingEnabled}
+                        bounces={false}
+                        onScroll={this.handleScroll}
+                        scrollEventThrottle={1}
+                        contentOffset={{x: (this.state.currentFile) * this.state.deviceWidth}}
+                    >
+                        {this.props.files.map((file) => {
+                            let component;
+                            if (file.has_preview_image) {
+                                component = (
+                                    <FileAttachmentPreview
+                                        file={file}
+                                        theme={this.props.theme}
+                                        imageHeight={Math.min(maxImageHeight, file.height)}
+                                        imageWidth={Math.min(this.state.deviceWidth, file.width)}
+                                        resizeMode='contain'
+                                        wrapperHeight={maxImageHeight}
+                                        wrapperWidth={this.state.deviceWidth}
+                                    />
+                                );
+                            } else {
+                                component = (
+                                    <FileAttachmentIcon
+                                        file={file}
+                                        theme={this.props.theme}
+                                        iconHeight={120}
+                                        iconWidth={120}
+                                        wrapperHeight={200}
+                                        wrapperWidth={200}
+                                    />
+                                );
+                            }
+
+                            return (
+                                <TouchableWithoutFeedback
+                                    key={file.id}
+                                    onPress={this.toggleFileInfo}
+                                >
+                                    <View style={[style.pageWrapper, {height: maxImageHeight, width: this.state.deviceWidth}]}>
+                                        {component}
+                                    </View>
+                                </TouchableWithoutFeedback>
+                            );
+                        })}
+                    </ScrollView>
+                    <AnimatedView
+                        style={[style.footerWrapper, {width: this.state.deviceWidth, opacity: this.state.footerOpacity}]}
+                        pointerEvents='none'
+                    >
+                        <LinearGradient
+                            style={style.footer}
+                            start={{x: 0.0, y: 0.0}}
+                            end={{x: 0.0, y: 0.9}}
+                            colors={['#0000', '#000000']}
```

Comment:
Sorry old habit, #0000 is actually transparent. I need to just start using `transparent`. I'll get it changed.